pandas/core/groupby/generic.py

```diff
@@ -1063,58 +1087,32 @@ def _cython_agg_blocks(
                     deleted_items.append(locs)
                     continue
                 else:
+                    if isinstance(result, Series):
```

Comment:
Hmm why is this required here?

Reply:
we have test cases that get here with a Series and without this the `cast` just below here is inaccurate

Reply:
Hmm that's unfortunate. Don't want to go too far down the typing road on this one but maybe should be using `assert ...` instead of `cast`

Reply:
cc @simonjayhawkins

Reply:
hmm, `reveal_type(result)` entering this block is object. probably need to take control of the type of result before this. does adding type annotations to  `_recast_result` help?

Reply:
it looks like since `aggregate` in `SelectionMixin` does not have type annotations, the return type of `s.aggregate(lambda x: alt(x, axis=self.axis))` is `Any` and hence as result is object no type narrowing occurs. reveal_type(s) is 'pandas.core.groupby.groupby.GroupBy' so it's just `aggregate` that needs the annotations.